src/ansys/motorcad/core/rpc_client_core.py

```diff
@@ -188,15 +193,19 @@ def __init__(
             else:  # port is not defined
                 self._find_free_motor_cad()
 
-        self.pid = self.get_process_id()
-
         # Check for response
         if self._wait_for_response(2) is True:
             self._connected = True
-            # Store Motor-CAD version number for any required backwards compatibility
 
+            # Store Motor-CAD version number for any required backwards compatibility
             self.program_version = self._get_program_version()
 
+            self.pid = self.get_process_id()
```

Comment:
Unrelated change but I think this was a mistake to not have this after the connection check

Reply:
I moved          

`self.pid = self.get_process_id()`

Reply:
This was in my initial commit but I think it might have been done in another PR too and merged from main so no longer showing.